src/main/java/com/agency04/sbss/pizza/rest/CustomerRestController.java

```diff
@@ -35,30 +32,32 @@ public Customer getCustomer(@PathVariable String username) {
 
     @DeleteMapping("/{username}")
     public ResponseEntity deleteCustomer(@PathVariable String username) {
-        Customer customer = customers.stream().filter(x -> x.getUserName().equals(username)).findFirst().orElse(null);
-        if(customer == null){
+        Customer tempCustomer = customerRepository.findByUsername(username);
+
+        if (tempCustomer == null) {
             throw new CustomerNotFoundException("Customer with username " + username + " doesn't exist.");
         }
-        customers.remove(customer);
+
+        customerRepository.deleteByUsername(username);
         return ResponseEntity.ok(HttpStatus.OK);
     }
 
     @PostMapping("/")
     public ResponseEntity registerCustomer(@RequestBody Customer customer) {
-        customers.add(new Customer(customer));
-        if(customers.stream().filter(x -> x.getCustomerId() == customer.getCustomerId()).findFirst().orElse(null) == null ){
+        customer.setId(0);
+        customerRepository.save(customer);
+        if(customerRepository.findByUsername(customer.getUsername()) == null ){
```

Comment:
Prvo spremiš novog customera pa zatim ideš u bazu i pitaš postoji li customer s tim usernameom. Ima li to smisla?